circleci/docker-publish

```diff
@@ -61,7 +71,7 @@ fi
 
 # dockerhub login.
 # `FROM image` in Dockerfile is going to pull from DockerHub
-docker login -u $DOCKER_USER -p $DOCKER_PASS --email="$DOCKER_EMAIL" || docker login -u $DOCKER_USER -p $DOCKER_PASS
+docker login -u $DOCKER_USER -p $DOCKER_PASS  || docker login -u $DOCKER_USER -p $DOCKER_PASS
```

Comment:
Is the duplicated command here as a form of retry logic?

Reply:
the reason both were there previously was to fallback depending on which version of Docker was running.

In the case that Docker version < X, email was a required arg, whereas for Docker version >= X, email was optional.

If email isn't ever required anymore, then we only need to run docker login one time.

Reply:
I'm not sure whether email is ever required, and I am not prioritizing looking into it right now. 
  Brought back this `--email` flag in f2f2d1196e694e985812c0e5b85b406665f11db6.